src/hangar/records/hashs.py

```diff
@@ -13,68 +20,178 @@ def __init__(self, hashenv: lmdb.Environment):
+        Union[bytes, Tuple[bytes, bytes]]
+            bytes of keys if ``keys=True`` and ``values=False``
+
+            bytes of values if ``keys=False`` and ``values=True``
+
+            Tuple of bytes corresponding to ``(keys, values)`` if
+            ``keys=True`` and ``values=True``.
         """
         startSchemaRangeKey = f'{c.K_SCHEMA}'.encode()
         try:
             hashtxn = TxnRegister().begin_reader_txn(self._hashenv)
             with hashtxn.cursor() as cursor:
                 schemasExist = cursor.set_range(startSchemaRangeKey)
-                while schemasExist:
-                    schemaRecKey, schemaRecVal = cursor.item()
-                    if schemaRecKey.startswith(startSchemaRangeKey):
-                        yield schemaRecKey
-                        schemasExist = cursor.next()
-                        continue
-                    else:
-                        schemasExist = False
+                # divide loop into returned type sections as perf optimization
+                # (rather then if/else checking on every iteration of loop)
+                if keys and not values:
+                    while schemasExist:
+                        schemaRecKey, schemaRecVal = cursor.item()
+                        if schemaRecKey.startswith(startSchemaRangeKey):
+                            yield schemaRecKey
+                            schemasExist = cursor.next()
+                            continue
+                        else:
+                            schemasExist = False
+                elif values and not keys:  # pragma: no cover
+                    while schemasExist:
+                        schemaRecKey, schemaRecVal = cursor.item()
+                        if schemaRecKey.startswith(startSchemaRangeKey):
+                            yield schemaRecVal
+                            schemasExist = cursor.next()
+                            continue
+                        else:
+                            schemasExist = False
+                elif keys and values:
+                    while schemasExist:
+                        schemaRecKey, schemaRecVal = cursor.item()
+                        if schemaRecKey.startswith(startSchemaRangeKey):
+                            yield (schemaRecKey, schemaRecVal)
+                            schemasExist = cursor.next()
+                            continue
+                        else:
+                            schemasExist = False
+                else:  # pragma: no cover
+                    raise ValueError(f'Both keys and values argument cannot be False')
         finally:
             TxnRegister().abort_reader_txn(self._hashenv)
 
     def list_all_hash_keys_raw(self):
-        recs = self._traverse_all_hash_records()
-        out = list(map(parsing.hash_data_raw_key_from_db_key, recs))
+        recs = self._traverse_all_hash_records(keys=True, values=False)
+        out = list(map(hash_data_raw_key_from_db_key, recs))
```

Comment:
list comprehension? Here and few other places where we have map -> list conversion

Reply:
list comprehension is ~10% slower than mapping when the function is predefined. I've changed it where appropriate, but won't be making the change universally.

Reply:
But since we have a conversion from map object to `list`, I don't think the performance gain map could bring is relevant anymore. It's going to be slower than list comprehension, IMO. I mean, it's probably negligible but just saying if you want to do a comparison